cmd/root.go

```diff
@@ -218,33 +204,32 @@ func (c *rootCommand) setupLoggers() (<-chan struct{}, error) {
 		loggerForceColors = !c.globalState.Flags.NoColor && c.globalState.Stdout.IsTTY
 		c.globalState.Logger.SetOutput(c.globalState.Stdout)
 	case line == "none":
-		c.globalState.Logger.SetOutput(ioutil.Discard)
-
+		c.globalState.Logger.SetOutput(io.Discard)
 	case strings.HasPrefix(line, "loki"):
+		c.loggerIsRemote = true
 		ch = make(chan struct{}) // TODO: refactor, get it from the constructor
-		hook, err := log.LokiFromConfigLine(c.globalState.Ctx, c.globalState.FallbackLogger, line, ch)
+		hook, err := log.LokiFromConfigLine(ctx, c.globalState.FallbackLogger, line, ch)
 		if err != nil {
-			return nil, err
+			close(ch)
```

Comment:
Can we do the refactor suggested by the TODO in this PR?

We currently [close the channel from the receiver side](https://github.com/grafana/k6/blob/0b563d4428391f6170d65bd30704b48b03a3f80f/log/loki.go#L165), which violates the [Channel Closing Principle](https://go101.org/article/channel-closing.html), and is generally a code smell.

With the additional closes here, we're relying on the channel not being closed already by the receiver to avoid the panic, which doesn't seem like it happens now, but it's best if we can avoid the uncertainty altogether.